### PR TITLE
feat: add cache client-side data doc

### PR DESCRIPTION
**Pull request by haijian-vaadin**

### CODE REVIEW

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+title: PWA cache client-side data
+order: 7
+layout: page
+---
+
+= Cache client-side data
+
+Vaadin caches the client-side views out-of-the-box, but not the client-side data. This may lead to the situation that even the client-side view is available offline, it's not very useful since there is no data. To overcome this, you can use https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage[localStorage] to store the client-side data, so that the data available even offline.
```
Comment (joheriks) [inline articles/fusion/pwa/pwa-cache-client-side-data.asciidoc:9]: Looks like linting is failing for some reason, but probably should adhere to the convention of one sentence per line

**Review by joheriks — CHANGES_REQUESTED** [review]

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+layout: page
+---
+
+= Cache client-side data
+
+Vaadin caches the client-side views out-of-the-box, but not the client-side data. This may lead to the situation that even the client-side view is available offline, it's not very useful since there is no data. To overcome this, you can use https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage[localStorage] to store the client-side data, so that the data available even offline. 
+
+== A cachable wrapper with localStorage
+
+Assume you have a function to retrieve the data from the backend. After the data is retrieved, you can save it into localStorage as a key-value pair. It's also good to provide a default value, in case the data is neither available from the backend nor cache.
```
Comment (joheriks) [inline articles/fusion/pwa/pwa-cache-client-side-data.asciidoc:13]: Maybe "local storage" or `localStorage` (formatted as code).

Comment (joheriks) [inline articles/fusion/pwa/pwa-cache-client-side-data.asciidoc:13]: Avoid contraction ("it's").

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+----
+// instead of always relying on the backend to retrive the data. e.g.
+// const stats = await getStats();
+// you can now wrap getStats into the cacheable helper.
+const stats = await cacheable(() => getStats(), 'stats', {});
+----
+
+== Clear the localStorage
+
+It's a good practice to clear the data stored in the localStorage, for example, when a user logs out. Suppose there is a `logout` event, the above cacheable wrapper could be improved as
```
Comment (joheriks) [inline articles/fusion/pwa/pwa-cache-client-side-data.asciidoc:51]: Avoid contraction "it's".

**Review by joheriks — CHANGES_REQUESTED** [review]

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc (hunk 1 above)
Comment (haijian-vaadin) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: done

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc (hunk 2 above)
Comment (haijian-vaadin) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: good catch, done

Comment (haijian-vaadin) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: done

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc (hunk 3 above)
Comment (haijian-vaadin) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: done

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+----
+// instead of always relying on the backend to retrive the data. e.g.
+// const stats = await getStats();
+// you can now wrap getStats into the cacheable helper.
+const stats = await cacheable(() => getStats(), 'stats', {});
+----
+
+== Clear the localStorage
+
+It is a good practice to clear the data stored in the localStorage, for example, when a user logs out.
```
Comment (joheriks) [inline articles/fusion/pwa/pwa-cache-client-side-data.asciidoc:55]: Could also be `localStorage`

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc (hunk 2 above)
Comment (joheriks) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: One more I initially missed, see comment below.

File: articles/fusion/pwa/pwa-cache-client-side-data.asciidoc (hunk 4 above)
Comment (haijian-vaadin) [reply articles/fusion/pwa/pwa-cache-client-side-data.asciidoc]: done

**Review by joheriks — APPROVED** [review]

**Pull request merged**